core/src/it/java/software/amazon/awssdk/internal/net/ConnectionUtilsIntegrationTest.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2010-2017. Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this filae. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.internal.net;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.getRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlMatching;
+import static java.util.Collections.emptyMap;
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.Inet4Address;
+import java.net.URI;
+import java.util.Collections;
+import java.util.HashMap;
+import org.junit.After;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class ConnectionUtilsIntegrationTest {
```

Comment:
Does this not work as a unit test?

Reply:
I usually consider it to be an integration test if it uses WireMock - since it actually does a network lookup here it's not strictly a unit-test

Reply:
Would it be good to have it run as part of build, similar to the protocol tests? Unless it takes a long time, of course.

Reply:
We mentioned it offline, but good to repeat here: we can call this a component test or something since it's not integrating with an external service and put it in the unit test directory.

Reply:
Renamed and moved under regular test/ folder